src/Main.ts

```diff
@@ -0,0 +1,876 @@
+
+    private slackHookHandler: SlackHookHandler;
+
+    // tslint:disable-next-line: no-any
+    private metrics: PrometheusMetrics;
+
+    private adminCommands = new AdminCommands(this);
+
+    constructor(public readonly config: IConfig) {
+        if (config.oauth2) {
+            this.oauth2 = new OAuth2({
+                client_id: config.oauth2.client_id,
+                client_secret: config.oauth2.client_secret,
+                main: this,
+                redirect_prefix: config.oauth2.redirect_prefix || config.inbound_uri_prefix,
+            });
+        }
+
+        const dbdir = config.dbdir || "";
+
+        this.bridge = new Bridge({
+            controller: {
+                onEvent: (request: Request) => {
+                    const ev = request.getData();
+                    this.stateStorage.onEvent(ev);
+                    this.onMatrixEvent(ev).then(() => {
+                        log.info(`Handled ${ev.event_id} (${ev.room_id})`);
+                    }).catch((ex) => {
+                        log.error(`Failed to handle ${ev.event_id} (${ev.room_id})`, ex);
+                    });
+                },
+                onUserQuery: () => ({}), // auto-provision users with no additional data
+            },
+            domain: config.homeserver.server_name,
+            eventStore: path.join(dbdir, "event-store.db"),
+            homeserverUrl: config.homeserver.url,
+            registration: "slack-registration.yaml",
+            roomStore: path.join(dbdir, "room-store.db"),
+            userStore: path.join(dbdir, "user-store.db"),
+        });
+
+        this.slackHookHandler = new SlackHookHandler(this);
+
+        if (config.enable_metrics) {
+            this.initialiseMetrics();
+        }
+    }
+
+    public getIntent(userId: string) {
+        return this.bridge.getIntent(userId);
+    }
+
+    public initialiseMetrics() {
+        this.metrics = this.bridge.getPrometheusMetrics();
+
+        this.bridge.registerBridgeGauges(() => {
+            const now = Date.now() / 1000;
+
+            const remoteRoomsByAge = new PrometheusMetrics.AgeCounters();
+            const matrixRoomsByAge = new PrometheusMetrics.AgeCounters();
+
+            this.rooms.forEach((room) => {
+                remoteRoomsByAge.bump(now - room.RemoteATime!);
+                matrixRoomsByAge.bump(now - room.MatrixATime!);
+            });
+
+            const countAges = (users: {[key: string]: MatrixUser|SlackGhost}) => {
+                const counts = new PrometheusMetrics.AgeCounters();
+
+                Object.keys(users).forEach((id) => {
+                    counts.bump(now - users[id].aTime!);
+                });
+
+                return counts;
+            };
+
+            return {
+                matrixRoomConfigs:
+                    Object.keys(this.roomsByMatrixRoomId).length,
+                remoteRoomConfigs:
+                    Object.keys(this.roomsByInboundId).length,
```

Comment:
both of these properties have weird line breaks: just put them on the line above. If you have to increase the line length limit, do it.